src/cdk-experimental/listbox/listbox.ts

```diff
@@ -7,98 +7,206 @@
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  constructor(private _el: ElementRef,
+              @Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {
+  }
 
   /** Toggles the selected state, emits a change event through the injected listbox */
   toggle() {
-    this.selected = !this.selected;
-    this.listbox._emitChangeEvent(this);
+    if (!this._getComboDisabled()) {
+      this.selected = !this.selected;
+      this.listbox._emitChangeEvent(this);
+    }
+  }
+
+  activateOption() {
+    if (!this._getComboDisabled()) {
+      this._active = true;
+      this.listbox.setActiveOption(this);
+    }
+  }
+
+  deactivateOption() {
+    this._active = false;
+  }
+
+  _getComboDisabled(): boolean {
+    return (this.listbox.disabled || this._disabled);
+  }
+
+  _getTabIndex(): string | null {
+    return (this.listbox.disabled || this._disabled) ? null : '-1';
+  }
+
+  getLabel(): string {
+    return this._el.nativeElement.textContent;
+  }
+
+  setActiveStyles() {
+    this._active = true;
+  }
+
+  setInactiveStyles() {
+    this._active = false;
   }
 
   static ngAcceptInputType_selected: BooleanInput;
+  static ngAcceptInputType_disabled: BooleanInput;
 }
 
-/**
- * Directive that applies interaction patterns to an element following the aria role of listbox.
- * Typically CdkOption elements are placed inside the listbox. Logic to handle keyboard navigation,
- * selection of options, active options, and disabled states is built in.
- */
 @Directive({
-  selector: '[cdkListbox]',
-  exportAs: 'cdkListbox',
-  host: {
-    role: 'listbox',
-  }
+    selector: '[cdkListbox]',
+    exportAs: 'cdkListbox',
+    host: {
+      role: 'listbox',
+      '(keydown)': '_keydown($event)',
+      '[attr.aria-disabled]': '_disabled',
+    }
 })
-export class CdkListbox {
+export class CdkListbox implements AfterContentInit, OnDestroy {
+
+  _listKeyManager: ActiveDescendantKeyManager<CdkOption>;
+  private _disabled: boolean = false;
 
-  /** A query list containing all CdkOption elements within this listbox */
   @ContentChildren(CdkOption, {descendants: true}) _options: QueryList<CdkOption>;
 
   @Output() readonly selectionChange: EventEmitter<ListboxSelectionChangeEvent> =
       new EventEmitter<ListboxSelectionChangeEvent>();
 
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  ngAfterContentInit() {
+    this._listKeyManager = new ActiveDescendantKeyManager(this._options)
+      .withWrap().withVerticalOrientation().withTypeAhead();
+  }
+
+  ngOnDestroy() {
+    this._listKeyManager.change.complete();
+  }
+
+  _keydown(event: KeyboardEvent) {
+    if (this._disabled) {
+      return;
+    }
+
+    const keyCode = event.keyCode;
+
+    switch (keyCode) {
```

Comment:
I would probably use an `if` statement rather than switch since there's only two cases (unless you plan to add more in a follow-up PR)

Reply:
I will likely add HOME and END. Even then since those two have the same functionality I can probably just put it all into if statements.

Reply:
I actually think I will just add the HOME and END functionality now since they aren't very complicated.